CONTRIBUTING.md

```diff
@@ -5,48 +5,84 @@ Want to contribute to Sitecore JavaScript Services? There are a few things you n
 
-* master - latest released version
-* dev - latest changes for the next release
-* release branches - created in order to make changes/updates to past major version releases
+To lint manually, the following commands are available from the root of the monorepo, you may also install the git-hook which will run these commands as a pre-push hook:
 
-# Documentation contributions
+```shell
+npm run lint-packages // lint everything under ./packages
+npm run lint-apps // lint everything under ./samples
+```
 
-If your documentation changes are related to a code feature that you are submitting to the JSS repo, you should make those changes in your code feature branch and submit them with your feature pull request.
+You may also lint a package by itself by running `npm run lint` while the `cwd` is the root of the package, for example:
 
-If your documentation changes are related to features that are _already available_ in JSS, then choose one of the approaches below.
+```shell
+cd ./packages/sitecore-jss
+npm run lint
+```
 
-## Edit on GitHub
+Some linting errors may be fixed automatically using the `--fix` flag. You may do this from the root, however you will need `--` twice before the flag in order to pass the argument to the inner command
 
-Every route on the documentation site (https://jss.sitecore.com) has a link at the bottom of the page titled `Edit this on GitHub`. Clicking this link will use the GitHub editing features to allow you to make changes to the page content.
+```shell
+npm run lint-packages -- -- --fix 
+```
 
-When your changes are complete, use the `Create a new branch for this comment and start a pull request` option. This will create a pull request to the `master` branch. Your changes will be reviewed and merged if appropriate.
+In addition to linting rules, there are a few coding guidelines worth mentioning here that will cause less friction when trying to get a PR merged.
+
+- TypeScript: make sure everything has the appropriate type.
+
+- JSDoc: new functions, interfaces, classes, etc. need to have a JSDoc style comment above their declaration. Example: https://jsdoc.app/howto-es2015-classes.html
+
+## Run unit tests
 
-## Edit in repo
+To keep everything running smoothly, please include unit tests when applicable.
+To run all tests, from the root of the monorepo:
 
-If you need to make code changes to the documentation site, follow these steps:
+```shell
+npm run test-packages
+```
+
+In the root of a package:
+
+```shell
+npm run test
+```
+
+# Troubleshooting
+
+Problem: Build of package or sample fails with the following error: `cannot find module ...`
+
+Solution: If dependencies in a package change, you may need to reset the repo to have the latest changes. This is often needed after pulling changes that include a dependency change.
```

Comment:
instead of always running a reset, shouldn't they do a bootstrap ?
and do a reset only if they want to clean the cache in node_modules.
Just saying because Lerna reset is time-consuming.

Reply:
True. My thought was that locally it would be rare to experience the 'cannot find module' error unless a reset was needed, but for us we see it all the time since we change branches often. I'll add that as the first option tho.